diff --git a/ports/vcpkg-tool-meson/meson-intl.patch b/ports/vcpkg-tool-meson/meson-intl.patch
index 6e2bcc31b..4b2eeb2dd 100644
--- a/ports/vcpkg-tool-meson/meson-intl.patch
+++ b/ports/vcpkg-tool-meson/meson-intl.patch
@@ -2,7 +2,7 @@
 +++ b/mesonbuild/dependencies/misc.py
 @@ -610,7 +610,8 @@ iconv_factory = DependencyFactory(
  
- intl_factory = DependencyFactory(
+ packages['intl'] = intl_factory = DependencyFactory(
      'intl',
 +    [DependencyMethods.BUILTIN, DependencyMethods.SYSTEM, DependencyMethods.CMAKE],
 +    cmake_name='Intl',
diff --git a/ports/vcpkg-tool-meson/portfile.cmake b/ports/vcpkg-tool-meson/portfile.cmake
index 27dfdf9a0..c29f8da9f 100644
--- a/ports/vcpkg-tool-meson/portfile.cmake
+++ b/ports/vcpkg-tool-meson/portfile.cmake
@@ -5,7 +5,7 @@
 set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
 
 set(program MESON)
-set(program_version 0.63.0)
+set(program_version 1.3.0)
 set(program_name meson)
 set(search_names meson meson.py)
 set(interpreter PYTHON3)
@@ -31,11 +31,11 @@ vcpkg_find_acquire_program(PYTHON3)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO mesonbuild/meson
-    REF bb91cea0d66d8d036063dedec1f194d663399cdf
-    SHA512 e5888eb35dd4ab5fc0a16143cfbb5a7849f6d705e211a80baf0a8b753e2cf877a4587860a79cad129ec5f3474c12a73558ffe66439b1633d80b8044eceaff2da
+    REF 7368795d13081d4928a9ba04d48498ca2442624b
+    SHA512 b2dc940a8859d6b0af8cb762c896d3188cadc1e65e3c7d922d6cb9a4ed7a1be88cd4d51a8aa140319a75e467816e714c409cf74c71c830bbc5f96bb81c1845ce
     PATCHES
         meson-intl.patch
-        remove-freebsd-pcfile-specialization.patch
+        #
 )
 
 vcpkg_execute_required_process(
diff --git a/ports/vcpkg-tool-meson/vcpkg.json b/ports/vcpkg-tool-meson/vcpkg.json
index a4c9b9a25..1de98470a 100644
--- a/ports/vcpkg-tool-meson/vcpkg.json
+++ b/ports/vcpkg-tool-meson/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "vcpkg-tool-meson",
-  "version": "0.63",
+  "version": "1.3.0",
   "port-version": 2,
   "description": "Meson build system",
   "homepage": "https://github.com/mesonbuild/meson",
